pkg/kubelet/kubelet.go

```diff
@@ -995,6 +995,30 @@ func (kl *Kubelet) podFieldSelectorRuntimeValue(fs *api.ObjectFieldSelector, pod
 	return fieldpath.ExtractFieldPathAsString(pod, internalFieldPath)
 }
 
+func (kl *Kubelet) lookupDnsService() (string, string, error) {
+	// Lookup the cluster DNS service and returns its info
+	if kl.serviceLister == nil {
+		return "", "", nil
+	}
+	services, err := kl.serviceLister.List()
```

Comment:
This seems like a very heavy way of doing single GET operation - am I missing something?